aws/protocol/query/array.go

```diff
@@ -63,10 +54,6 @@ func (a *Array) Value() Value {
 
 	// Query lists start a 1, so adjust the size first
 	a.size++
-	prefix := a.prefix
-	if !a.flat {
-		prefix = fmt.Sprintf("%s.%s", prefix, a.memberName)
-	}
 	// Lists can't have flat members
-	return newValue(a.values, fmt.Sprintf("%s.%d", prefix, a.size), false)
+	return newValue(a.values, fmt.Sprintf("%s.%d", a.prefix, a.size), false)
```

Comment:
Why leave Sprintf here?

Reply:
The extra int formatting for this case causes an increase in allocations when the size of the array being serialized was between 100-500. There's a nice drop in CPU but a lot of the CPU utilization we see is driven by GC so I chose to leave this one alone. I'll edit the full benchmark results in to the PR description.